### PR TITLE
[encointer] introduce democracy and community treasuries

### PR DESCRIPTION
**Pull request by brenzi**
Over the past few months, Encointer has developed a democracy pallet with one-person-one-vote governance for local communities. We think we're ready to experiment with our real communities on mainnet.

The governance scope of local democracy is restricted to 

* properties of the local community itself (like amount of [basic community-issued income](https://book.encointer.org/economics-ubi.html))
* spending of its own treasury funds (currently restricted to KSM). see [towards sustainability](https://forum.polkadot.network/t/towards-encointer-self-sustainability/4195)
* petitions (`system.remark`)

Therefore, no special `Origin` is granted which could affect the KSM token or other parts of Kusama. 

In contrast to token voting on *OpenGov*, Encointer democracy aims at **one-person-one-vote**, enabled by its sybil-resilient proof-of-personhood protocol. In practice, it will be one-cycle-attendance-one-vote, which is a pretty decent approximation with some additional weight given to those who participate in the protocol more actively

Relevant pallet code
* https://github.com/encointer/pallets/tree/master/democracy
   * work partially funded by a [w3f grant](https://grants.web3.foundation/applications/democratic-governance-1)
* https://github.com/encointer/pallets/tree/master/treasuries / [initial MVP PR](https://github.com/encointer/pallets/pull/401)

Design documents:
* https://book.encointer.org/protocol-democracy.html
* initial forum post introducing the design: https://forum.polkadot.network/t/encointers-grassroots-democracy-prototype/9009

Testing on Testnet Gesell
* https://book.encointer.org/tutorials-democracy.html
* https://book.encointer.org/tutorials-treasuries.html

Testing with bot communities (check CI logs)
* https://github.com/encointer/encointer-node/pull/370

### CODE REVIEW

File: system-parachains/encointer/src/lib.rs [hunk 1]
```diff
@@ -545,6 +545,29 @@ impl pallet_encointer_faucet::Config for Runtime {
...
+	pub const ConfirmationPeriod: Moment = 2 * 24 * 3600 * 1000; // [ms]
+	pub const ProposalLifetime: Moment = 9 * 24 * 3600 * 1000; // [ms]
+}
+
+impl pallet_encointer_democracy::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type MaxReputationCount = ConstU32<64>;
+	type ConfirmationPeriod = ConfirmationPeriod;
+	type ProposalLifetime = ProposalLifetime;
+	type MinTurnout = ConstU128<50>; // permill of electorate: 1 = 0.1%, 50 = 5.0%
```
Comment (joepetrowski) [inline system-parachains/encointer/src/lib.rs:558]: Wouldn't 5% of a permill be `50_000`?

File: system-parachains/encointer/src/lib.rs [hunk 2]
```diff
@@ -788,6 +813,7 @@ mod benches {
 		[pallet_encointer_bazaar, EncointerBazaar]
 		[pallet_encointer_ceremonies, EncointerCeremonies]
 		[pallet_encointer_communities, EncointerCommunities]
+		[pallet_encointer_democracy, EncointerDemocracy]
```
Comment (joepetrowski) [inline system-parachains/encointer/src/lib.rs:816]: No treasuries weights?

**Review by joepetrowski — APPROVED** [review]

File: system-parachains/encointer/src/lib.rs (hunk 2 above)
Comment (brenzi) [reply system-parachains/encointer/src/lib.rs:816]: treasuries has no dispatchables, therefore no explicit benchmarking

File: system-parachains/encointer/src/lib.rs (hunk 1 above)
Comment (brenzi) [reply system-parachains/encointer/src/lib.rs:558]: good catch. actually, the confusion lies deeper. this setting is correct as-is according to the pallet implementation which treats the value as perthousand. I'll adjust the naming:
https://github.com/encointer/pallets/blob/43e099dce05f44ce3d4d945117b43099d5718aa1/democracy/src/lib.rs#L559-L562

**Review by acatangiu — APPROVED** [review]
Looks good!

**Pull request merged**